openassessment/fileupload/api.py

```diff
@@ -173,15 +210,63 @@ class FileUploadManager(object):
     """
     def __init__(self, openassessment_xblock):
         self.block = openassessment_xblock
+        self._cached_shared_uploads = None
 
     def get_uploads(self):
         """
         Returns:
             A list of FileUpload objects associated with an instance of an O.A. Block.
+            This will include FileUpload objects corresponding to existing files
+            *and* files that have been deleted.
```

Comment:
Are you sure? I'm pretty sure by default _file_uploads_from_list_fields ignores deleted files.

Reply:
oh, hmm, I reasoned myself into this to convince myself that we didn't have a bug in the file display template, but now I think we might